docs/proposals/storage_secrets.md

```diff
@@ -0,0 +1,94 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+In Kubernetes, the API object 'secrets' provide a means to store confidential credentials in a volume to be accessed by 
+the Kuberenetes Master as well as other external resources for the pod, such as database login information. 
+
+Creating a secret volume for the storage layer has use cases outside the norm of traditional access.  Today, the
+access to the secret is typically contrainted to a namepsace and application within that namespace requiring the information.
+Secrets are created for a purpose that can be isolated by one function alone. Storage security presents the obstacle that secrets 
+with actual storage assets may not have a specific namespace when the volumes are created. Therefore, facilitiating the need for an
+alternate approach.
+
+## Motivation
+
+Today, ACLs are not well defined in storage. An admin cannot create precise contraints for a user or group based on the actual storage
+assets and have it flow through the system. In addition, administrators whom pre-create volumes to expedite the process or to limit users, don't know which user or group may want to request that resource. Since secrets today are namespace constrainted, we cannot attach the
+secret globally. Nor, would that solution necessarily be desireable considering the possibility of exposure.
+
+There are specific functions, unique to storage that require the credentials to perform different actions. Today, when performing an 
+attach/detatch the node must have this information to perform this action. This would entail copying the secret information from the 
```

Comment:
> Attach detach controller in the master pattern only works with a cloud provider (or hypervisor) which is able to force a computer to do something from the outside.  

This can be fixed pretty easily.  If your cluster needs to use iSCSI or FCoE, then you run an "attach helper" on each node (e.g. addon DaemonSet).    It supports an RPC API to attach or detach disks.  It authenticates its caller as being the the attach controller.   The attach controller only needs to send the helper the secrets it needs to use for that particular attach operation.